src/hostapi/coreaudio/pa_mac_core.c

```diff
@@ -2136,29 +2136,35 @@ static OSStatus ringBufferIOProc(
     AudioStreamPacketDescription** outDataPacketDescription,
     void*                          inUserData)
 {
-   UInt32 * ioDataSize = &ioData->mBuffers[0].mDataByteSize;
-   void ** outData = &ioData->mBuffers[0].mData;
+   VVDBUG(("ringBufferIOProc()\n"));
 
-   void *dummyData;
-   ring_buffer_size_t dummySize;
    PaUtilRingBuffer *rb = (PaUtilRingBuffer *) inUserData;
 
-   VVDBUG(("ringBufferIOProc()\n"));
-
    if( PaUtil_GetRingBufferReadAvailable( rb ) == 0 ) {
-      *outData = NULL;
-      *ioDataSize = 0;
+      ioData->mBuffers[0].mData = NULL;
+      ioData->mBuffers[0].mDataByteSize = 0;
+      *ioNumberDataPackets = 0;
```

Comment:
Returning a value through *ioNumberDataPackets is new behavior. Was that part of the bug?

Reply:
absolutely yes!